File: Backend/src/main/java/com/web/matcha/web/dto/ResearchMembersDto.java
package com.web.matcha.web.dto;

import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
public class ResearchMembersDto {

	/**
	 * The min age of the members.
	 */
	private Integer ageMin;

	/**
	 * The max age of the members.
	 */
	private Integer ageMax;

	/**
	 * The min fame rating of the members.
	 */
	private Integer fameRatingMin;

	/**
	 * The max distance of the members.
	 */
	private Integer distanceMax;

	/**
	 * The interests of the members.
	 */
	private List<String> interests;
}

File: Frontend/src/app/data/dto/send/research-members.dto.ts
/**
 * The dto for the research members request
 */
export class ResearchMembersDto {

	/**
	 * The number of the page.
	 */
	page?: number;

	/**
	 * The min age of the members.
	 */
	ageMin?: number;

	/**
	 * The max age of the members.
	 */
	ageMax?: number;

	/**
	 * The min fame rating of the members.
	 */
	fameRatingMin?: number;

	/**
	 * The max distance of the members.
	 */
	distanceMax?: number;

	/**
	 * The interests of the members.
	 */
	interests?: string[];
}